src/services/account.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Bytes } from '../model/general.types'
+import { MemoryAccount } from '../model/storage/account.model'
+import { Network } from '../model/storage/network.model'
+import {
+  bytesToWordArray,
+  decrypt,
+  encrypt,
+  hexToWordArray,
+  wordArrayToBytes,
+  wordArrayToHex,
+} from '../utils/encryption'
+import { Storage } from './storage/storage.service'
+
+export class AccountService {
+  private storage: Storage = new Storage()
+
+  public async create(
+    name: string,
+    address: string,
+    password: string,
+    seed: Bytes<64>,
+    network: Network,
+  ): Promise<void> {
+    const account = await this.storage.getAccount(name)
+
+    if (account) {
+      throw new Error('Account already exists')
+    }
+
+    await this.storage.setAccount({
+      name,
+      address,
+      network,
+      seed: this.encryptSeed(seed, password),
+    })
+  }
+
+  public async load(name: string, password: string): Promise<MemoryAccount> {
+    const account = await this.storage.getAccount(name)
+
+    if (!account) {
+      throw new Error("Account doesn't exists")
+    }
+
+    const memoryAccount: MemoryAccount = account as unknown as MemoryAccount
+
+    memoryAccount.seed = this.decryptSeed(account.seed, password)
```

Comment:
I don't see here you would check whether the password is correct or not.
As we discussed, it can be done by checking the ethereum address derived from seed is matching with the saved ethereum address

Reply:
That logic has been moved to the auth.listener:

https://github.com/fairDataSociety/blossom/blob/b4eeba2267b3381d7e61ad2bbeda51d03130e9ee/src/listeners/message-listeners/auth.listener.ts#L50-L60

The fdp-storage must be initialized with the seed in order to get address.